modified_cdc_protocol/operation_types/setup_pcr_plate/protocol.rb

```diff
@@ -54,25 +58,41 @@ def main
       default_operation_params: default_operation_params
     )
     
-
     provision_plates(
       operations: operations
     )
-
-    operations.retrieve
-
-    record_lot_numbers(operations: operations)
-
+    
+    create_microtiter_plates(operations: operations)
+    all_inputs = build_stripwell_primer_probe_compositions(operations: operations)
+    
+    if operations.errored.any?
+       error_operations
+       return {}
+    end
+    
     show_prepare_workspace
-
-    build_stripwell_master_mix_compositions(operations: operations)
-
+    
+    set_locations(all_inputs, 'Bench')
+    
+    retrieve_materials(all_inputs)
+    
+    record_lot_numbers(operations: operations)
+    
     assemble_primer_probe_plates(operations: operations)
 
     operations.store
 
     {}
   end
+  
+  def error_operations
```

Comment:
We should converge on a single method for doing this.

Reply:
c.f. method in Prepare RT-qPCR Plate, possibly others

Reply:
The issue I have with just "return {}" as in "Prepare RT-qPCR Plate" is it then shows the operation as "done" unless you explicitly set the status to something else.    When if it indeed ended early it shouldn't be done....     Here I set all the statuses to "pending" then return {}...

Reply:
Yes, that's a good point. I will take a shot at writing a common method that does all the good things.